pkg/provider/azure_loadbalancer.go

```diff
@@ -2381,7 +2381,7 @@ func (az *Cloud) getExpectedLBRules(
 	// healthcheck proxy server serves http requests
 	// https://github.com/kubernetes/kubernetes/blob/7c013c3f64db33cf19f38bb2fc8d9182e42b0b7b/pkg/proxy/healthcheck/service_health.go#L236
 	var nodeEndpointHealthprobe *network.Probe
-	if servicehelpers.NeedsHealthCheck(service) {
+	if servicehelpers.NeedsHealthCheck(service) && !consts.IsPLSProxyProtocolEnabled(service.Annotations) {
```

Comment:
Can we also check if service.beta.kubernetes.io/azure-pls-create is true? want to make sure pls is indeed needed here?

Reply:
updated. Thanks!